milton-processor/cache.ts

```diff
@@ -4,14 +4,37 @@ import base64url = require('base64url');
 import { ENVIRONMENT, CONFIG } from './scribe';
 
 
-export interface Manuscript {
-  url: string,
+export interface ManuscriptData {
+  url: string;
   title: string;
   siteName: string;
   byline: string;
   excerpt: string;
   textContent: string;
   content: string;
+  cachedAt: number;
+  updatedAt: number;
+}
+
+export class Manuscript {
+  data: ManuscriptData;
+
+  constructor(data: ManuscriptData) {
+    this.data = data;
+  }
+
+  updateUpdatedAt() {
+    this.data.updatedAt = Date.now();
+  }
+
+  static fromJson(jsonData: string): Manuscript {
+    const data = JSON.parse(jsonData) as ManuscriptData;
+    return new Manuscript(data);
+  }
+
+  toJson(): string {
+    return JSON.stringify(this.data);
+  }
```

Comment:
This works, but is not very idiomatic; I think the standard way to customize JSON serialization is to have a `toJSON` (note different case) method which returns a JSON structure and is [automatically called by `JSON.stringify`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON_behavior).